apps/src/blockly/addons/cdoTrashcan.js

```diff
@@ -2,6 +2,52 @@ import GoogleBlockly from 'blockly/core';
 import {ToolboxType} from '../constants';
 
 export default class CdoTrashcan extends GoogleBlockly.Trashcan {
+  /**
+   * @override
+   */
+  init() {
+    super.init();
+    this.workspace_.addChangeListener(this.workspaceChangeHandler.bind(this));
+  }
+
+  workspaceChangeHandler(blocklyEvent) {
+    if (blocklyEvent.type === Blockly.Events.BLOCK_DRAG) {
+      let trashcanDisplay = 'none';
+      let toolboxVisibility = 'visible';
+      const isDraggingFromFlyout_ = !!Blockly.mainBlockSpace?.currentGesture_
+        ?.flyout_;
+      if (!isDraggingFromFlyout_ && blocklyEvent.isStart) {
+        trashcanDisplay = 'block';
+        toolboxVisibility = 'hidden';
+      }
+
+      /**
+       * NodeList.forEach() is not supported on IE. Use Array.prototype.forEach.call() as a workaround.
+       * https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach
```

Comment:
ah yes I've run into this issue with IE before. Thanks for adding the comment + link!